### DISCUSSION TITLE
Smart table typescript error

### DISCUSSION THREAD

**Discussion by fiqryq** [coreui/coreui#372] [Q&A] [answered]
hello i want to ask something , how to use smart table on typescript react? i use smart table on my coreui pro using typescript and keep getting error like this when I try to custom table button and badge
![Screen Shot 2022-03-14 at 22 40 12](https://user-images.githubusercontent.com/25787603/158207956-e7c0d5b7-676f-4d1e-b272-9b00762cc8d2.png)

**Comment by mrholek** [comment, marked as answer]
@fiqryq What's the error message?

**Reply by fiqryq** [reply]
i got error like this : 
![Screen Shot 2022-03-14 at 22 46 46](https://user-images.githubusercontent.com/25787603/158209142-bfca6d32-55dd-481d-9424-271d15ee1d97.png)

**Reply by fiqryq** [reply]
here's my columns code :  

```js
  const columns = [
    {
      key: "id",
      _style: { width: "40%" },
      filter: false,
      label: "No",
    },
    {
      key: "data_catalog_detail",
      _style: { width: "20%" },
      filter: false,
      label: "Nama Katalog yang direquest",
    },
    {
      key: "requested_by",
      _style: { width: "20%" },
      filter: false,
      label: "Peminta Data",
    },
    {
      key: "request_detail",
      _style: { width: "20%" },
      filter: false,
      label: "Alasan Permintaan",
    },
    {
      key: "requested_at",
      _style: { width: "20%" },
      filter: false,
      label: "Tanggal",
    },
    {
      key: "request_status",
      _style: { width: "20%" },
      filter: false,
      label: "Status",
    },
    {
      key: "aksi",
      _style: { width: "20%" },
      filter: false,
      label: "Aksi",
    },
  ];
```

**Reply by mrholek** [reply]
Which version of `@coreui/react-pro` do you have installed?

**Reply by fiqryq** [reply]
i use this : 
```
"@coreui/coreui-pro": "4.0.1",
"@coreui/react-pro": "4.0.0-rc.0",
```

**Reply by mrholek** [reply]
You have to update both to 4.1.4

**Reply by fiqryq** [reply]
already update but still get same error
